google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -0,0 +1,502 @@
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": e.type_,
+                    "original_text": e.mention_text,
+                }
+            )
+        elif e.provenance.type_ == _OP_TYPE.REMOVE:
+            entity = entities_array.pop(int(e.id) - 1)
+            history.append(
+                {
+                    "object": "Entity",
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": entity.type_,
+                    "original_text": entity.mention_text,
+                }
+            )
+            del entity
+        elif e.provenance.type_ == _OP_TYPE.REPLACE:
+            history.append(
+                {
+                    "object": "Entity",
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": e.type_,
+                    "original_text": e.mention_text,
+                    "replace_type": e.type_,
+                    "replace_text": e.mention_text,
+                }
+            )
+            entities_array[int(e.id) - 1] = e
+    return entities_array, history
+
+
+def _get_level(current_revision: "DocumentWithRevisions") -> int:
+    r"""Returns the level of the current revision in the revision tree.
+
+    Args:
+        current_revision (DocumentWithRevisions):
+            The current revision.
+
+    Returns:
+        int:
+            The level of the current revision in the revision tree.
+    """
+    try:
+        level = 0
+        p = current_revision.parent
+        if p.revision_id:
+            while p:
+                p = p.parent
+                level += 1
+        return level
+    except Exception:
+        return 0
+
+
+def _print_child_tree(
+    current_revision: "DocumentWithRevisions",
+    doc: "DocumentWithRevisions",
+    seen: List[str],
+):
+    r"""Prints the revision tree of a child revision.
+
+    Args:
+        None.
+
+    Returns:
+        None.
+    """
+    tab = "  " * _get_level(doc)
+
+    if doc.children:
+        if current_revision.revision_id == doc.revision_id:
+            print(tab + "└──>", end="")
+        else:
+            print(tab + "└──", end="")
+        print(doc.revision_id)
+        for each in doc.children:
+            seen.append(each.revision_id)
+            _print_child_tree(current_revision, each, seen)
+    else:
+        if current_revision.revision_id == doc.revision_id:
+            print(tab + "└──>", end="")
+        else:
+            print(tab + "└──", end="")
+        print(doc.revision_id)
+
+
+@dataclasses.dataclass
+class DocumentWithRevisions:
+    r"""Represents a wrapped Document.
+
+    A single Document protobuf message with revisions will written as several `dp.bp` files on
```

Comment:
"A single Document protobuf message with revisions" in the docstring here suggests that an instance of this is a protobuf message (which it is not) that comes with special attributes such as `revisions`.  But from the presence of the `next_`, `parent` attributes, we seem to be saying an instance of this class _is_ a revision (that is, a single document at a particular point in time)?  These are quite different kinds of data - what are we modeling here?